lib/default.nix

```
@@ -78,4 +78,59 @@ rec {
     import ./macvtap.nix {
       inherit microvmConfig hypervisorConfig lib;
     };
+
+  /*
+    extractOptValues - Extract and remove all occurrences of a command-line option and its values from a list of arguments.
+
+    Description:
+      This function searches for a specified option flag in a list of command-line arguments,
+      extracts ALL associated values, and returns both the values and a filtered list with
+      all occurrences of the option flag and its values removed. The order of all other
+      arguments is preserved. Uses tail recursion to process the argument list.
+
+    Parameters:
+      optFlag :: String | [String] - The option flag(s) to search for. Can be:
+                                     - A single string (e.g., "-platform")
+                                     - A list of strings (e.g., ["-p" "-platform"])
+                                     All matching flags and their values are extracted
+      extraArgs :: [String] - A list of command-line arguments
+
+    Returns:
+      {
+        values :: [String] - List of all values associated with matching flags (empty list if none found)
+        args :: [String] - The input list with all matched flags and their values removed
+      }
+
+    Examples:
+      # Extract single occurrence:
+      extractOptValues "-platform" ["-vnc" ":0" "-platform" "linux" "-usb"]
+      => { values = ["linux"]; args = ["-vnc" ":0" "-usb"]; }
+
+      # Extract multiple occurrences:
+      extractOptValues "-platform" ["-a" "a" "-b" "b" "-c" "c" "-b" "b2"]
```

```suggestion
      extractOptValues "-b" ["-a" "a" "-b" "b" "-c" "c" "-b" "b2"]
```

If I understand this correctly, the example should have ``-b`` instead.